tests/fragments/test_google_protobuf_type.proto

```diff
@@ -0,0 +1,49 @@
+// Copyright (C) 2023  Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
```

Comment:
The idea here is that this proto replicates the essential features of servicemanagement that were causing the failure due to an older test runtime? If so, could you

(1) Add a comment here to that effect, so it's clear why we're adding this fragment test
(2) Verify manually that simply adding this file at the current HEAD (before merging this PR) does cause a test failure. I want to ensure this test really captures the current error condition and then the fix you've implemented. (You may already have done this, in which case that's great!)

Reply:
1) Done in [2a7f516](https://github.com/googleapis/gapic-generator-python/pull/1749/commits/2a7f516d9e3ab0a4e288b90b4a84e83f600a44ac)
2) I've verified that adding this file fails without the fix for the default templates. The ads templates don't support constraints files yet. I filed https://github.com/googleapis/gapic-generator-python/issues/1788 to tackle that in a future iteration as it is non-trivial.